app/components/DocsLayout.tsx

```diff
@@ -12,27 +17,193 @@ import { useLocalStorage } from '~/utils/useLocalStorage'
 import { DocsCalloutQueryGG } from '~/components/DocsCalloutQueryGG'
 import { DocsCalloutBytes } from '~/components/DocsCalloutBytes'
 import { DocsLogo } from '~/components/DocsLogo'
-import type { DocsConfig } from '~/utils/config'
+import { generatePath, last } from '~/utils/utils'
+import type { AvailableOptions } from '~/components/Select'
+import type { ConfigSchema, MenuItem } from '~/utils/config'
 
-export function DocsLayout({
-  name,
-  version,
-  colorFrom,
-  colorTo,
-  textColor,
+/**
+ * Use framework in URL path
+ * Otherwise use framework in localStorage if it exists for this project
+ * Otherwise fallback to react
+ */
+function useCurrentFramework(frameworks: AvailableOptions) {
+  const { framework: paramsFramework } = useParams()
+  const localStorageFramework = localStorage.getItem('framework')
+
+  return paramsFramework
+    ? paramsFramework
+    : localStorageFramework
+    ? frameworks[localStorageFramework]
+      ? localStorageFramework
+      : 'react'
+    : 'react'
```

Comment:
wdyt? (You don't have to change this if you don't want to, it's just a suggestion!)
```suggestion
    return (
    paramsFramework ||
    (localStorageFramework && localStorageFramework in frameworks
      ? localStorageFramework
      : 'react')
    )

```

Reply:
Yeah that's definitely nicer 👍